extensions/vscode/src/multiStepInputs/selectConfig.ts

```diff
@@ -135,7 +137,9 @@ export async function selectConfig(
     async (resolve, reject) => {
       try {
         const python = await getPythonInterpreterPath();
-        const inspectResponse = await api.configurations.inspect(python);
+        const inspectResponse = await api.configurations.inspect(python, {
+          dir: activeDeployment.projectDir,
+        });
```

Comment:
This is selecting a config for an existing deployment, so we have to limit choices to the configs which are present in that particular project dir where the entrypoint was found.